src/env.rs

```diff
@@ -234,4 +236,16 @@ impl Environment {
         self.surfaces.lock().unwrap().push(surface.clone());
         surface
     }
+
+    /// Create a new shell surface
+    pub fn create_shell_surface<Impl>(
+        &self,
+        surface: &Proxy<wl_surface::WlSurface>,
+        shell_impl: Impl,
+    ) -> Arc<Box<ShellSurface>>
```

Comment:
I don't think the wrapping into an `Arc` needs to be done at this level, this was an implementation detail of `Window`.